clipmenu

```diff
@@ -33,27 +42,34 @@ fi
 
 # Blacklist of non-dmenu launchers
 launcher_args=(-l "${CM_HISTLENGTH}")
-if [[ "$CM_LAUNCHER" == fzf ]]; then
-    launcher_args=()
-fi
-
-# rofi supports dmenu-like arguments through the -dmenu flag
-[[ "$CM_LAUNCHER" == rofi ]] && set -- -dmenu "$@"
 
-list_clips() {
-    LC_ALL=C sort -rnk 1 < "$cache_file" | cut -d' ' -f2- | awk '!seen[$0]++'
-}
+if (( CM_INPUT_CLIP )); then
+    read
+    chosen_line="$REPLY"
+else
+    # Blacklist of non-dmenu launchers
+    launcher_args=(-l "${CM_HISTLENGTH}")
+    if [[ "$CM_LAUNCHER" == fzf ]]; then
+        launcher_args=()
+    fi
 
-if [[ "$CM_LAUNCHER" == rofi-script ]]; then
-    if (( $# )); then
-        chosen_line="${!#}"
-    else
+    # rofi supports dmenu-like arguments through the -dmenu flag
+    [[ "$CM_LAUNCHER" == rofi ]] && set -- -dmenu "$@"
+
+    if [[ "$CM_LAUNCHER" == rofi-script ]]; then
+        if (( $# )); then
+            chosen_line="${!#}"
+        else
+            list_clips
+            exit
+        fi
+    elif [[ "$CM_LAUNCHER" == cat ]]; then
```

Comment:
Hmm, why not just use the same logic as fzf and actually use `CM_LAUNCHER`?

Reply:
You lost me, that's what I did?

Also, could you review the other PR I did, which has, under my view, a much better solution to this problem.